### PR TITLE
add option to trim uuid and not prefix

### PR DESCRIPTION
**Pull request by helgridly**
**PR checklist**
- [ ] For each library you've modified here, decide whether it requires a major, minor, or no version bump. (Click [here](/broadinstitute/workbench-libs/blob/develop/CONTRIBUTING.md) for guidance)

If you're doing a **major** or **minor** version bump to any libraries:
- [ ] Bump the version in `project/Settings.scala` `createVersion()`
- [ ] Update `CHANGELOG.md` for those libraries
- [ ] I promise I used `@deprecated` instead of deleting code where possible

In all cases:
- [ ] Get two thumbsworth of PR review
- [ ] Verify all tests go green (CI _and_ coverage tests)
- [ ] Squash commits and **merge to develop**
- [ ] Delete branch after merge

After merging, _even if you haven't bumped the version_: 
- [ ] Update `README.md` and the `CHANGELOG.md` for any libs you changed with the new dependency string. The git hash is the same short version you see in GitHub (i.e. seven characters). You can commit these changes straight to develop.

### CODE REVIEW

File: model/src/main/scala/org/broadinstitute/dsde/workbench/model/google/package.scala [hunk 1]
```diff
@@ -28,29 +28,36 @@ package object google {
...
 
-    // max length of 63 chars, including the uuid
     val uuid = UUID.randomUUID.toString
-    val maxNameLength = 63 - uuid.length - 1
-    if (sb.length > maxNameLength) sb.setLength(maxNameLength)
+
+    //if we're trimming the prefix, we make the prefix short enough to accommodate the UUID.
+    //if we're trimming the UUID, we might still need to trim the prefix anyway if it's enormous.
+    val trimmedPrefix = if(trimPrefix) sb.take(maxBucketNameLen - uuid.length) else sb.take(maxBucketNameLen)
+    val trimmedUUID = if(trimPrefix) uuid else sb.take(Math.min(maxBucketNameLen - trimmedPrefix.length, uuid.length))
```
Comment (jmthibault79) [inline model/src/main/scala/org/broadinstitute/dsde/workbench/model/google/package.scala]: Shouldn't this be `uuid.take()` ?

File: model/src/test/scala/org/broadinstitute/dsde/workbench/model/google/GcsPathParserSpec.scala [hunk 2]
```diff
@@ -20,6 +20,15 @@ class GcsPathParserSpec extends FlatSpecLike with Matchers with EitherValues {
     generateUniqueBucketName("myClusterWhichHasAVeryLongNameBecauseIAmExtremelyVerboseInMyDescriptions").value.length shouldBe 63
   }
 
+  "gcs" should "generate valid bucket names when trimming the uuid" in {
```
Comment (jmthibault79) [inline model/src/test/scala/org/broadinstitute/dsde/workbench/model/google/GcsPathParserSpec.scala:23]: add test case for true

Comment (jmthibault79) [reply model/src/test/scala/org/broadinstitute/dsde/workbench/model/google/GcsPathParserSpec.scala:23]: never mind, default is true

File: model/src/main/scala/org/broadinstitute/dsde/workbench/model/google/package.scala (hunk 1 above)
Comment (helgridly) [reply model/src/main/scala/org/broadinstitute/dsde/workbench/model/google/package.scala]: how did my tests not catch this

File: model/src/test/scala/org/broadinstitute/dsde/workbench/model/google/GcsPathParserSpec.scala (hunk 2 above)
Comment (helgridly) [reply model/src/test/scala/org/broadinstitute/dsde/workbench/model/google/GcsPathParserSpec.scala]: the default is true, it's the block above

**Pull request merged**